src/Kubernetes/KubernetesApiExtensions.cs

```diff
@@ -503,6 +503,31 @@ public static V1Job WithBillingId(this V1Job job, string billingId)
             });
         }
 
+        /// <summary>
+        /// Adds a policy failure action and exit codes to the job.
+        /// </summary>
+        /// <param name="job">The job object to modify.</param>
+        /// <param name="actionExitCodeId">A list with all action names and corresponding exit code number.</param>
+        /// <returns>The Kubernetes Job object with added pod failure policy rules.</returns>>
+        public static V1Job WithPodPolicyFailureExitCodes(this V1Job job, List<(string action, int exitCode)> actionExitCodeId)
+        {
+            var groupedActions = actionExitCodeId.GroupBy(aec => aec.action);
```

Comment:
not sure why do you need to group? If you want to avoid the case when people add the same stuff twice, use `Dictionary<string, List<int>>` instead

Reply:
That is definitely a good idea! 
I've updated it now in [3822912](https://github.com/SneaksAndData/esd-services-sdk/pull/75/commits/38229123666d315f956b5f23e8cf98a9a1a445d5)
Thank you for the suggestion 🙂